Dungeoneer/src/com/interrupt/dungeoneer/ui/layout/GameMessage.java

```diff
@@ -0,0 +1,107 @@
+package com.interrupt.dungeoneer.ui.layout;
+
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.ui.Label;
+import com.badlogic.gdx.scenes.scene2d.ui.VerticalGroup;
+import com.badlogic.gdx.utils.Align;
+import com.interrupt.dungeoneer.game.Game;
+import com.interrupt.dungeoneer.overlays.OverlayManager;
+import com.interrupt.dungeoneer.ui.UiSkin;
+
+public class GameMessage extends Element {
+    @Override
+    protected Actor createActor() {
+        VerticalGroup group = new VerticalGroup();
+
+        group.addActor(createMessageLabel());
+        group.addActor(createTitleLabel());
+        group.addActor(createAttributesLabel());
+
+        group.debug();
+        group.columnAlign(Align.center);
+        group.space(4f);
+
+        return group;
+    }
+
+    private Label createMessageLabel() {
+        Label label = new Label("MESSAGE", UiSkin.getSkin()) {
+            @Override
+            public void act(float delta) {
+                super.act(delta);
+
+                if (Game.message2.hasActiveMessage()) {
+                    setText(Game.message2.getMessage());
+                    setColor(Color.WHITE);
+                    setVisible(true);
+                }
+                else {
+                    setText("");
+                    setVisible(false);
+                }
+            }
+        };
+
+        label.debug();
+
+        return label;
+    }
+
+    private Label createTitleLabel() {
+        Label label = new Label("TITLE", UiSkin.getSkin()) {
+            @Override
+            public void act(float delta) {
+                super.act(delta);
+
+                if (!shouldShowUseMessage()) {
+                    setText("");
+                    setVisible(false);
+                    return;
+                }
+
+                setText(Game.instance.player.lookedAtItem.getLookAtInfo().getTitle());
+                setColor(Game.instance.player.lookedAtItem.getLookAtInfo().getTitleColor());
```

Comment:
+1 to newing up a DTO here as well. Basically anywhere you access `lookedAtItem.getLookAtInfo()` you are creating new objects.

Could the info potentially change? Either way, I think we can be a bit more careful with regards to GC. 🗑️

Reply:
Yes, info can change when e.g. a door opens or closes. But it is now cached in the player.